recipes/openssl/3.x.x/conanfile.py

```diff
@@ -420,6 +420,11 @@ def generate(self):
             env.define_path("CROSS_SDK", os.path.basename(xcrun.sdk_path))
             env.define_path("CROSS_TOP", os.path.dirname(os.path.dirname(xcrun.sdk_path)))
 
+        if is_apple_os(self):
```

Comment:
```suggestion
        if is_apple_os(self) and self.options.shared:
```

I would apply this flag only when building shared libraries, because it's destined for dylib only.

Reply:
Thank you for this suggestion. I applied in my latest commit (8d60991).